nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/TestMonitorActivity.java

```diff
@@ -349,6 +408,50 @@ public void testFirstRunNoMessages() throws InterruptedException {
         while(rerun);
     }
 
+    @Test
+    public void testRunNoMessagesWithWaitForActivityTrue() throws InterruptedException {
+        // don't use the TestableProcessor, we want the real timestamp from @OnScheduled
+        final TestRunner runner = TestRunners.newTestRunner(new MonitorActivity());
+        runner.setProperty(MonitorActivity.CONTINUALLY_SEND_MESSAGES, "false");
+        runner.setProperty(MonitorActivity.THRESHOLD, "5 secs");
+        runner.setProperty(MonitorActivity.WAIT_FOR_ACTIVITY, "true");
+
+        // shouldn't generate inactivity b/c run() will reset the lastSuccessfulTransfer if @OnSchedule & onTrigger
+        // does not  get called more than MonitorActivity.THRESHOLD apart
+        runner.run();
+        runner.assertTransferCount(MonitorActivity.REL_SUCCESS, 0);
+        runner.assertTransferCount(MonitorActivity.REL_INACTIVE, 0);
+        runner.assertTransferCount(MonitorActivity.REL_ACTIVITY_RESTORED, 0);
+        Thread.sleep(10000L);
+        runNext(runner);
+        runner.assertTransferCount(MonitorActivity.REL_SUCCESS, 0);
+        runner.assertTransferCount(MonitorActivity.REL_INACTIVE, 0);
+        runner.assertTransferCount(MonitorActivity.REL_ACTIVITY_RESTORED, 0);
+        runner.clearTransferState();
+    }
+
+    @Test
+    public void testRunNoMessagesWithWaitForActivityFalse() throws InterruptedException {
+        // don't use the TestableProcessor, we want the real timestamp from @OnScheduled
+        final TestRunner runner = TestRunners.newTestRunner(new MonitorActivity());
+        runner.setProperty(MonitorActivity.CONTINUALLY_SEND_MESSAGES, "false");
+        runner.setProperty(MonitorActivity.THRESHOLD, "5 secs");
+        runner.setProperty(MonitorActivity.WAIT_FOR_ACTIVITY, "false");
+
+        // shouldn't generate inactivity b/c run() will reset the lastSuccessfulTransfer if @OnSchedule & onTrigger
+        // does not  get called more than MonitorActivity.THRESHOLD apart
+        runner.run();
+        runner.assertTransferCount(MonitorActivity.REL_SUCCESS, 0);
+        runner.assertTransferCount(MonitorActivity.REL_INACTIVE, 0);
+        runner.assertTransferCount(MonitorActivity.REL_ACTIVITY_RESTORED, 0);
+        Thread.sleep(10000L);
+        runNext(runner);
+        runner.assertTransferCount(MonitorActivity.REL_SUCCESS, 0);
+        runner.assertTransferCount(MonitorActivity.REL_INACTIVE, 1);
+        runner.assertTransferCount(MonitorActivity.REL_ACTIVITY_RESTORED, 0);
+        runner.clearTransferState();
+    }
+
```

Comment:
The sleep of 10 seconds is too long for efficient unit testing. The other unit test method provides some basic coverage, so recommend removing these two test methods.